README.md

```diff
@@ -4,3 +4,112 @@
 # MQTT bridge for Apache Kafka®
 
 This project provides a software component which acts as a bridge between [MQTT 3.1.1](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html) and an [Apache Kafka®](https://kafka.apache.org/) cluster.
+
+### Running Tests
+
+```bash
+mvn clean test
+```
+
+## MQTT bridge documentation
+
+This section provides information about the MQTT Bridge. It describes the MQTT Bridge, how it works, and how to configure it.
+
+### 1. MQTT Bridge Overview
+
+#### 1.1 How it works
+
+To enable a seamless integration between MQTT and Kafka, the MQTT Bridge provides a way to map MQTT topics to Kafka topics. 
+This mapping is done using a set of predefined patterns and kafka topic templates. The MQTT Bridge uses these patterns to map MQTT topics to Kafka topics. 
+As a part of the bridge, a Kafka producer will be responsible for producing messages from the MQTT clients to the Kafka Cluster.
+
+#### 1.2. Topic Mapping Rules(TOMAR)
+
+The TOMAR is a set of patterns the user provides defining how the MQTT Bridge maps MQTT topic names to Kafka topic names.
+A Mapping Rule is a model that contains an `MQTT topic pattern` and a `Kafka topic template`. 
+All the incoming MQTT message's topic should match an `MQTT topic pattern` in the `TOMAR` so that the bridge knows in which Kafka topic to produce this message. This Kafka topic is defined by a template, `kafka Topic template`. 
+However, if the incoming MQTT message's topic does not match any pattern in the `TOMAR`, the Bridge has default Kafka topic where the incoming message will be mapped to, know as `messages_default`.
+
+A valid TOMAR is a JSON file that contains an array of mapping rules. 
+Each mapping rule is a JSON object that contains two properties: `mqttTopic` and `kafkaTopic`. The following is an example of a valid TOMAR:
+
+```json
+[
+  {
+    "mqttTopic": "building/{building}/room/{room}/#",
+    "kafkaTopic": "building_{building}_room_{room}"
+  },
+  {
+    "mqttTopic": "sensors/#",
+    "kafkaTopic": "sensor_others"
+  },
+  {
+    "mqttTopic": "sensors/+/data",
+    "kafkaTopic": "sensor_data"
+  }
+]
+```
+The wildcard "#" represents one or more levels in the topic hierarchy. The wildcard "+" represents a single level in the topic hierarchy.
```

Comment:
```suggestion
The wildcard "#" represents one or more levels in the topic hierarchy. 
The wildcard "+" represents a single level in the topic hierarchy.
```

Mentioning the placeholder is missing. Also explain that `#` can stay only at the end.
Finally a link to the MQTT spec explaining this would be useful to users.